### DISCUSSION TITLE
Load supportFile from node_modues?

### DISCUSSION THREAD

**Discussion by pschaubroeck** [cypress-io/cypress#25311] [Questions and Help] [answered]
Is it possible to load a supportFile from node_modules?  i've been trying to create a cross project cypress configuration where all cypress setup is obfuscated away into a separate package, utilizing the node api to do so.  And I keep getting the error below.

For reference, before anyone asks why, my end goal is a complete quality tools like package.  linting/formatting/testing/etc that is all pre-configured for upwards of 50-150+ isolated packages.  i've found a ton of duplicate setup required as is, and want to pull it all out of the way so devs don't need to think or worry about it.  There will be aspects of configurability but that will be very opinionated due to our overall vision.

```
Your supportFile is missing or invalid: node_modules/<path to custom support file>/support.js

The supportFile must be a .js, .ts, .coffee file or be supported by your preprocessor plugin (if configured).

Fix your support file, or set supportFile to false if a support file is not necessary for your project.

If you have just renamed the extension of your supportFile, restart Cypress.

https://on.cypress.io/support-file-missing-or-invalid
```

the same configuration works fine in the obfuscated package, but not when i import it into a library that is to consume it.  I have verified the path is correct. It's a little strange that the reported path is relative, when i supplied an absolute path to the support file.  This makes me think it's somewhat related to the fileServerFolder or some other configuration i'm missing.  Any help would be appreciated.

Alternatively, if there is a simple way to add a command (and all it's functionality) that doesn't involve utilizing supportFile, i'm all ears.  I have yet to find anything referencing that possibility however.

**Comment by emilyrohrbough** [comment, marked as answer]
Please see: https://github.com/cypress-io/cypress/issues/23616

**Reply by pschaubroeck** [reply]
thanks @emilyrohrbough!
